kong/clustering/compat/checkers.lua

```diff
@@ -23,6 +24,53 @@ end
 
 
 local compatible_checkers = {
+  { 3007000000, --[[ 3.7.0.0 ]]
+    function(config_table, dp_version, log_suffix)
+      local has_update
+
+      local flavor = kong and
+                     kong.configuration and
+                     kong.configuration.router_flavor
+      -- remove this once the `expressions` flavor supports `route.snis`
+      if flavor == "expressions" then
```

Comment:
I think that snis with wildcard also can not work with traditional route lower than 3.7.
In 3.6 and others the traditional router can not recognize snis like "*.com" or "a.*",
so this logic of flavor is needless.

Reply:
Will remove this once your refactor PR is merged. Currently `route.snis` doesn't exist for the `expressions` flavor.

Reply:
Indeed.